src/main/java/cmput402/tdd/App.java

```diff
@@ -158,18 +158,75 @@ public void editShoppingList(Scanner input, ShoppingList shoppingList){
         }
 
     }
+    public void editRecipe(Scanner input, Recipe recipe) throws Exception{
+        clearConsole();
+        System.out.println("Please select an option to edit your recipe:");
+        System.out.println("command format (ignore bracket): (add 1) (remove 5) (rename \"apples\")");
+        System.out.println("*add\n*remove\n*rename\n");
+        Integer quantity;
+        Item item;
+        String param;
+        String command = input.nextLine();
+        /*
+        Eg:
+        command = add 5
 
-    public Recipe createRecipe(Scanner input) throws Exception {
-        String name = input.nextLine();
+        function will split command into variables action and param
+        action = add
+        quantity = 5
+         */
+        Integer spaceIndex = command.indexOf(" ");
+        if(spaceIndex == -1){
+            throw new Exception("Invalid command format");
+        }
+        String action = command.substring(0, spaceIndex);
 
-        if (name.length() == 0){
-            throw new Exception("Name cannot be empty.");
+        if(action.equals("add") || action.equals("remove")) {
+            param = command.substring(spaceIndex + 1, command.length());
+            try{
+                quantity = Integer.parseInt(param);
+            }
+            catch(Exception e){
+                throw new Exception("Please enter a positive whole number for quantity");
+            }
+            try{
+                item = createItem(input);
+            }
+            catch (Exception e) {
+                throw new Exception("Failed to create item for recipe");
+            }
+            if(action.equals("add")){
+                if(quantity > 0){
+                    recipe.add(item, quantity);
+                    System.out.println("Added "+param+" "+item.getName()+" to recipe successfully");
+                }
+                else{
+                    throw new Exception("Quantity must be above 0");
+                }
+
+            }
+            else if(action.equals("remove")){
+                if(quantity >= 0 && recipe.items.get(item) >= quantity){
+                    recipe.remove(item, quantity);
+                    System.out.println("Removed "+param+" "+item.getName()+" from recipe successfully");
+                }
+                else{
+                    throw new Exception("Could not remove item from recipe");
```

Comment:
Error message should tell the user how to fix it to make it work

Reply:
Changed to more meaningful commit message